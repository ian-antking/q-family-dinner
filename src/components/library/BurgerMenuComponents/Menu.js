import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { bool } from 'prop-types';

export const StyledMenu = styled.nav`
  position: absolute;
  z-index: 2;
  top: 50px;
  right: 0;
  height: calc(100vh - 60px - 40px);
  width: 260px;
  padding: 20px 0;
  background: whitesmoke;
  border-top: 1px solid #E6E6E6;
  text-align: center;
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: flex-end;
  transform: ${({ open }) => (open ? 'translateX(0)' : 'translateX(100%)')};
  transition: transform 0.3s ease-in-out;
`;

const MenuLink = styled(Link)`
  width: calc(100% - 40px);
  height: 75px;
  padding: 0 40px 0 0;
  font-size: 1.8em;
  color: white;
  text-decoration: none;
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-end;
  align-items: center;

  &:hover {
    color: black;
  }
`;

const Menu = ({ open, setOpen, content }) => {
  const pages = content.pages;

  return (
    <StyledMenu data-testid="menu" open={open}>
      {pages && pages.map(page => { 
        return page.fields.primaryPage && (
          <MenuLink
            key={`menulink-${page.sys.id}`}
            to={`/${page.fields.slug}`}
            onClick={() => setOpen(!open)}
          >
            {page.fields.title}
          </MenuLink>
        );
      })
}

    </StyledMenu>
  );
};

Menu.propTypes = {
  open: bool.isRequired,
};

export default Menu;
