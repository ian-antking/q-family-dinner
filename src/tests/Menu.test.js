import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Menu from '../components/library/BurgerMenuComponents/Menu';

const closed = false;
const open = true;

test('snapshot: Menu component', () => {
	const { asFragment } = render(
		<Router>
	  		<Menu open={closed} />
	  	</Router>
	);
	const component = asFragment();
	expect(component).toMatchSnapshot();
  });

  test('Menu component renders', () => {
	const { getByTestId } = render(
	  <Router>
		<Menu open={closed} />
	  </Router>
	);
	const component = getByTestId('menu');
	expect(component).toBeInTheDocument();
  });
