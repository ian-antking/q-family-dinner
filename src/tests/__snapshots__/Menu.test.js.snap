// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: Menu component 1`] = `
<DocumentFragment>
  <nav
    class="sc-AxjAm fyOvYq"
    data-testid="menu"
  >
    <a
      class="sc-AxirZ bematR"
      href="#/"
    >
      Home
    </a>
    <a
      class="sc-AxirZ bematR"
      href="#/"
    >
      Also Home
    </a>
    <a
      class="sc-AxirZ bematR"
      href="#/"
    >
      Home Again!
    </a>
  </nav>
</DocumentFragment>
`;
